src/matrix/matrix.jl

```diff
@@ -131,14 +179,130 @@ function ==(M::smatrix{T}, N::smatrix{T}) where T <: AbstractAlgebra.RingElem
    return Bool(libSingular.mp_Equal(M.ptr, N.ptr, R.ptr))
 end
 
+###############################################################################
+#
+#   Concatenation
+#
+###############################################################################
+
+@doc Markdown.doc"""
+   hcat(A::smatrix{T}, B::smatrix{T}) where T <: AbstractAlgebra.RingElem
+> Return the horizontal concatenation of $A$ and $B$.
+> Assumes that the number of rows is the same in $A$ and $B$.
+"""
+function hcat(A::smatrix{T}, B::smatrix{T}) where T <: AbstractAlgebra.RingElem
```

Comment:
These don't seem very Singular specific. We should look into why the generic methods in AbstractAlgebra aren't working here.

Reply:
I remove them from my next update.